src/tools/sfdk/command.cpp

```diff
@@ -59,7 +62,309 @@ const char SDK_MAINTENANCE_TOOL[] = "SDKMaintenanceTool.app/Contents/MacOS/SDKMa
+                *errorString = valueTooBigMessage();
+                return false;
+            }
+            return true;
+        } else if (name == VM_CPU_COUNT) {
+            if (!parsePositiveInt(&m_cpuCount, value, errorString))
+                return false;
+            if (m_cpuCount > VirtualMachine::availableCpuCount()) {
+                *errorString = valueTooBigMessage();
+                return false;
+            }
+            return true;
+        } else if (name == VM_VDI_CAPACITY_MB) {
+            if (!parsePositiveInt(&m_vdiCapacityMb, value, errorString))
+                return false;
+            if (m_vdiCapacityMb < m_vm->vdiCapacityMb()) {
+                *errorString = valueCannotBeDecreasedMessage();
+                return false;
+            }
+            return true;
+        } else {
+            *errorString = unknownPropertyMessage(name);
+            return false;
+        }
+    }
+
+    bool set() override
+    {
+        bool ok = true;
+
+        if (m_memorySizeMb != m_vm->memorySizeMb()) {
+            bool stepOk;
+            execAsynchronous(std::tie(stepOk), std::mem_fn(&VirtualMachine::setMemorySizeMb),
+                    m_vm.data(), m_memorySizeMb);
+            ok &= stepOk;
+        }
+
+        if (m_cpuCount != m_vm->cpuCount()) {
+            bool stepOk;
+            execAsynchronous(std::tie(stepOk), std::mem_fn(&VirtualMachine::setCpuCount),
+                    m_vm.data(), m_cpuCount);
+            ok &= stepOk;
+        }
+
+        if (m_vdiCapacityMb != m_vm->vdiCapacityMb()) {
+            bool stepOk;
+            execAsynchronous(std::tie(stepOk), std::mem_fn(&VirtualMachine::setVdiCapacityMb),
+                    m_vm.data(), m_vdiCapacityMb);
+            ok &= stepOk;
+        }
+
+        return ok;
+    }
+
+private:
+    QPointer<VirtualMachine> m_vm;
+    int m_memorySizeMb = 0;
+    int m_cpuCount = 0;
+    int m_vdiCapacityMb = 0;
+};
+
+/*!
+ * \class EmulatorPropertiesAccessor
+ */
+
+class EmulatorPropertiesAccessor : public PropertiesAccessor
+{
+public:
+    EmulatorPropertiesAccessor(Emulator *emulator)
+        : m_emulator(emulator)
+        , m_vmAccessor(std::make_unique<VirtualMachinePropertiesAccessor>(
+                    emulator->virtualMachine()))
+    {
+    }
+
+    QMap<QString, QString> get() const override
+    {
+        return m_vmAccessor->get();
+    }
+
+    bool prepareSet(const QString &name, const QString &value, bool *needsVmOff,
+            QString *errorString) override
+    {
+        return m_vmAccessor->prepareSet(name, value, needsVmOff, errorString);
+    }
+
+    bool set() override
+    {
+        return m_vmAccessor->set();
+    }
+
+private:
+    QPointer<Emulator> m_emulator;
+    std::unique_ptr<VirtualMachinePropertiesAccessor> m_vmAccessor;
+};
+
+/*!
+ * \class BuildEnginePropertiesAccessor
+ */
+
+class BuildEnginePropertiesAccessor : public PropertiesAccessor
```

Comment:
To me it looks like BuildEnginePropertiesAccessor and EmulatorPropertiesAccessor are identical except for the contained type (Emulator vs BuildEngine). Consider using a template?